vnext/Microsoft.ReactNative/Views/TextViewManager.cpp

```diff
@@ -47,6 +53,13 @@ class TextShadowNode final : public ShadowNodeBase {
         transformableText.originalText = text;
         text = transformableText.TransformText();
         textBlock.Text(winrt::hstring(text));
+
+        if (m_isHighlighted) {
+          AddHighlighter(m_ColorValue, text.size());
+        }
+
+        m_prevCursorEnd += textBlock.Text().size();
+
```

Comment:
is this guaranteed to run "in the right order"?

Reply:
So we do have the 'index' argument in AddView(child, index) but I've debugged this code, and it always gets called in increasing order of the indices. Also, our logic of kind of already depends on it - you can see that if there's only one Run, there's a separate case for idx==0 above, and if there're more, we erase TextBlock.Text() and add it as an inline along with the incoming others during construction.

Reply:
Confirmed at UIManagerModule.cpp@205 and 209:
> // 1) Sort the views to add and indices to remove by index
> // 3) Iterate the views being added by index low to high and add